operations/kaas/cluster-stack-operator.md

```diff
@@ -0,0 +1,692 @@
+
+Here are some key points about Kubernetes events:
+
+1. **Purpose**: They give a chronological account of what's happening in the system. This can be crucial for debugging purposes, for understanding the lifecycle of resources, or for monitoring.
+2. **Scope**: Events can be associated with different types of objects. For instance, you might see events related to a Pod's lifecycle or events showing reasons why a service endpoint was not created.
+3. **Viewing Events**: The most common way to view events is by using the `kubectl` command-line tool:
+    - `kubectl get events`: Lists all events in the current namespace.
+    - `kubectl get events -n <namespace>`: Lists events in the specified namespace.
+    - `kubectl describe <resource-type> <resource-name>`: Provides detailed information about a specific resource, including a list of related events.
+4. **Event Types**: Kubernetes events can be of different types. Common ones include:
+    - **Normal**: Routine events that inform about standard operations.
+    - **Warning**: These events indicate something went wrong or might go wrong, and the user should take note.
+5. **Event Sources**: The `Source` of an event tells you which component generated the event. This can be a component of the Kubernetes system like `kubelet` or `scheduler`, or it can be the name of a server where an event occurred.
+6. **Reason**: Each event provides a `Reason` field which is a short, machine-readable string that gives a unique description of the event.
+7. **Message**: Along with the reason, an event contains a `Message` field which gives a human-readable description of the event.
+8. **Event Aggregation**: To avoid spamming the event log with the same event, Kubernetes aggregates similar consecutive events. You might see messages like “This event has been seen X times in the last Y minutes” for frequently recurring events.
+9. **Event Exporters**: There are tools and systems like Prometheus and Eventrouter that can be set up to scrape or receive these events and use them for monitoring, alerting, or logging purposes.
+
+In practice, Kubernetes events are a vital tool for developers and operators to understand what's going on inside their cluster, diagnose issues, and monitor the health and behavior of their applications and resources.
+
+We believe that events are, together with the conditions in the status of the event, the two sources that users should know in order to understand the system. Events might show some form of development of a certain incident over time, e.g. that the user understands that something has been going wrong 10 minutes ago and that is why the system is still stabilizing, but that the problem is already gone. 
+
+The conditions, on the other hand, should always show the current information that is available to describe the state of a resource.
+
+### Controllers that react immediately on specific etcd events
```

Comment:
`etcd events` -> `API server updates`?

(`etcd` should be abstracted away, `events` may be ambiguous with Kubernetes Event Resources)